www/js/lib/uiUtil.js

```diff
@@ -43,34 +43,87 @@ define(rqDef, function(settingsStore) {
      * @param {String} nodeAttribute The attribute to set to the URI
      * @param {Uint8Array} content The binary content to convert to a URI
      * @param {String} mimeType The MIME type of the content
-     * @param {Function} callback An optional function to call with the URI
+     * @param {Function} callback An optional function to call to start processing the next item
      */
     function feedNodeWithBlob(node, nodeAttribute, content, mimeType, callback) {
         // Decode WebP data if the browser does not support WebP and the mimeType is webp
         if (webpMachine && /image\/webp/i.test(mimeType)) {
+            // If we're dealing with a dataURI, first convert to Uint8Array
+            if (/^data:/i.test(content)) {
+                // Below uses Mozilla-recommended alternative to atob() : see https://developer.mozilla.org/en-US/docs/Glossary/Base64
+                content = util.dataURItoUint8Array(content);
+                // Or we could use webpHero's utility: it uses deprecated atob() but probably OK for image data
+                // content = webpHero.convertDataURIToBinary(content);
```

Comment:
There is a choice here between using the Mozilla-recommended alternative to deprecated atob() (which I've included in util.js, in place of a deprecated function that was there), or using the wepHero conversion utility. The webpHero version does use atob(), but as it will never be dealing with UTF8 data (it only deals with WebP images), this is probably OK.